olive/evaluator/olive_evaluator.py

```diff
@@ -415,81 +413,83 @@ def get_inference_settings(cls, metric: Metric, model: ONNXModelHandler) -> Dict
         if model_infrerence_settings:
             inference_settings.update(model_infrerence_settings)
 
-        metric_inference_settings = metric.get_inference_settings(cls.framework.lower())
+        metric_inference_settings = metric.get_inference_settings(Framework.ONNX.lower())
         if metric_inference_settings:
             inference_settings.update(metric_inference_settings)
 
         return inference_settings
 
-    def _inference(
-        self,
+
+class OnnxEvaluator(OliveEvaluator, OnnxEvaluatorMixin, framework=Framework.ONNX):
+    def __init__(self):
+        super().__init__()
+
+    @staticmethod
+    def get_session_wrapper(
         model: ONNXModelHandler,
         metric: Metric,
         dataloader: Dataset,
-        post_func=None,
-        device: Device = Device.CPU,
-        execution_providers: Union[str, List[str]] = None,
-    ) -> Tuple[OliveModelOutput, Any]:
+        device: Device,
```

Comment:
Should we add `device_id`?
We may need to reconsider about the interface to expose `device_id` in which components.

Reply:
same as above